bitnami/thanos/templates/storegateway/tls-server-secret.yaml

```
@@ -0,0 +1,13 @@
+{{- if and (.Values.storegateway.grpc.tls.enabled) (not .Values.storegateway.grpc.tls.existingSecret) }}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ include "common.secrets.name" (dict "existingSecret" .Values.storegateway.grpc.tls.existingSecret "defaultNameSuffix" "store-tls-server" "context" $) }}
```

I understand using this complex macro in the statefulset, but in this manifest you've already checked (1st line) that you need to create the secret (instead of reusing an existing one). Therefore, you can make it simple:
```suggestion
  name: {{ printf "%s-store-tls-server" (include "common.names.fullname" .) | trunc 63 | trimSuffix "-" }}
```
(feel free to ignore this suggestion)